util/types/time_test.go

```diff
@@ -587,6 +587,9 @@ func (s *testTimeSuite) TestRoundFrac(c *C) {
 		{"2012-12-31 11:30:45.123456", 1, "2012-12-31 11:30:45.1"},
 		{"2012-12-31 11:30:45.999999", 4, "2012-12-31 11:30:46.0000"},
 		{"2012-12-31 11:30:45.999999", 0, "2012-12-31 11:30:46"},
+		{"2012-00-00 11:30:45.999999", 3, "2012-00-00 11:30:46.000"},
+		// MySQL can handle this case, but we can't.
```

Comment:
This is not a big problem, it must satisfy conditions:
1. time round overflow one day, so it must be 23.59.59.9999999
2. date part invalid, contains month = 0 or day =0

The case is so rare that may be it won't be done in the future ...